src/Type/Definition/NonNull.php

```diff
@@ -4,13 +4,25 @@
 
 namespace GraphQL\Type\Definition;
 
+use GraphQL\Error\InvariantViolation;
+use GraphQL\Type\Schema;
+use function assert;
+use function is_callable;
+
 class NonNull extends Type implements WrappingType, OutputType, InputType
 {
-    /** @var NullableType&Type */
+    /** @var callable|(NullableType&Type) */
     private $ofType;
 
-    public function __construct(NullableType $type)
+    /**
+     * code sniffer doesn't understand this syntax. Pr with a fix here: waiting on https://github.com/squizlabs/PHP_CodeSniffer/pull/2919
+     * phpcs:disable
+     * @param  (NullableType&Type)|callable $type
+     */
+    public function __construct($type)
     {
+        assert($type instanceof NullableType || is_callable($type), new InvariantViolation('NonNull constructor expects an instance of NullableType, or a callable'));
```

Comment:
This can be dropped, should be guarded by PHPStan

Reply:
48106ac